code/drasil-utils/lib/Data/Map/Ordered.hs

```diff
@@ -0,0 +1,39 @@
+module Data.Map.Ordered (OrderedMap,
+    null, size, empty, singleton,
+    insert, lookup, elems, fromList) where
+
+import Prelude hiding (lookup, null)
+
+import qualified Data.Map as M
+import Data.List (sortOn)
+import Data.Foldable (Foldable(foldl'))
+
+-- | Simple ordered map. Based on 'container's 'Data.Map'. Note: Deletions are
+-- unsupported.
+newtype OrderedMap k v = OM { im :: M.Map k (Int, v) }
+  deriving (Show, Eq)
```

Comment:
I just wrote this instead of using an external library. I think we have something similar in our `ChunkDB` -- though I don't remember why we need to recall order of insertion of things into the `ChunkDB`, but if this is merged, we can use this one as well.

This was necessary for the rendering of basis blades to be done in order of insertion.